src/headers.rs

```diff
@@ -112,8 +112,28 @@ impl TryFrom<&[u8]> for Headers {
         };
 
         if let Some(line) = lines.next() {
-            if !line.starts_with("NATS/") {
-                return parse_error("version line does not begin with NATS/");
+            let mut parts = line.splitn(3, ' ');
+
+            if let Some(v) = parts.next() {
+                if !v.starts_with("NATS") {
+                    return parse_error("version line does not begin with NATS/");
+                }
+            }
+
+            if let Some(v) = parts.next() {
```

Comment:
Most of the time will not have inline status, so be good not to spend cycles when not needed. Hence the top level check on line length in the other PR.

Reply:
Yeah I ended up with more or less your take in the end minus one allocation.